loadgen/test_settings_internal.cc

```diff
@@ -109,12 +118,22 @@ TestSettingsInternal::TestSettingsInternal(
     constexpr double kSlack = 1.1;
     int target_sample_count =
         kSlack * DurationToSeconds(target_duration) * target_qps;
-    samples_per_query = std::max<int>(min_query_count, target_sample_count);
+    samples_per_query = 
+        (requested.performance_issue_unique || requested.performance_issue_same) 
+	     ? performance_sample_count
+	     : std::max<int>(min_query_count, target_sample_count);
     min_query_count = 1;
     target_duration = std::chrono::milliseconds(0);
   }
 
   min_sample_count = min_query_count * samples_per_query;
+
+  // Validate TestSettings
+  assert(requested.performance_issue_same_index <
+         performance_sample_count);
+  assert(!(requested.performance_issue_unique &&
+           requested.performance_issue_same));
```

Comment:
Since assert will only apply to debug builds, could you also log an error?

if (condition) {
  LogDetail([](AsyncDetail& detail) {
        detail.Error("message");
  }
}